sockeye/data_io.py

```diff
@@ -1261,24 +1286,29 @@ def create_sequence_readers(sources: List[str], targets: List[str],
                             vocab_sources: List[vocab.Vocab],
                             vocab_targets: List[vocab.Vocab]) -> Tuple[List[SequenceReader], List[SequenceReader]]:
     """
-    Create source readers with EOS and target readers with BOS.
+    Create source readers with EOS and target readers with BOS if vocabularies are passed.
 
     :param sources: The file names of source data and factors.
     :param targets: The file name of the target data and factors.
     :param vocab_sources: The source vocabularies.
     :param vocab_targets: The target vocabularies.
     :return: The source sequence readers and the target reader.
     """
+    # if not vocab_sources or not vocab_targets:
+    #     source_sequence_readers = [SequenceReader(source, None, add_eos=False) for source in sources]
+    #     target_sequence_readers = [SequenceReader(target, None, add_bos=False) for target in targets]
+    # else:
     source_sequence_readers = [SequenceReader(source, vocab, add_eos=True) for source, vocab in
-                               zip(sources, vocab_sources)]
+                                zip(sources, vocab_sources)]
     target_sequence_readers = [SequenceReader(target, vocab, add_bos=True) for target, vocab in
-                               zip(targets, vocab_targets)]
+                                zip(targets, vocab_targets)]
     return source_sequence_readers, target_sequence_readers
 
 
 def parallel_iter(source_iterables: Sequence[Iterable[Optional[Any]]],
                   target_iterables: Sequence[Iterable[Optional[Any]]],
-                  skip_blanks: bool = True):
+                  skip_blanks: bool = True,
+                  binary: bool = False):
```

Comment:
I'd still rename. I'd even consider adding a flag `check_token_parallel` which we could set to False for sharding and to True later when we load the shard (in the spirit of keeping the sharding logic minimal).

Reply:
Yes, I added that flag because when the file was being read in binary mode the check whether the tokens were parallel failed because the length of binary data did not match.

With the flag for check_token_parallel set to False when sharding then we wouldn't ensure that the tokens are parallel though. I guess it's fine since parallelism is checked later on. If we don't want to check token parallelism when sharding then the `binary` flag I added can be removed.